### DISCUSSION TITLE
More granular port naming. Adding calculations to naming. vMX example.

### DISCUSSION THREAD

**Discussion by ivanovuri** [GNS3/gns3-gui#3413] [Ideas]
Hi,
This question already [raised](https://gns3.com/community/featured/advanced-port-name-formatting) few years ago.
Looks like everyone agreed that it will be very coll feature to create more granular naming.

I just want to propose adding some sort of calculations. IMHO this would be easier than changing GUI. Mo' better to create editor like in default switch menu of course.

I.e. we all using vMX where first two ports are special purpose ports. What about adding calculation to make third port ge-0/0/0?
ge-0/0/{port0-1} = ge-0/0/-1, ge-0/0/0

**Comment by eantowne** [comment]
If you create a custom appliance file you can actually specify the name of
every port.

On Wed, Dec 14, 2022 at 02:56 ivanovuri ***@***.***> wrote:

> Hi,
> This question already raised
> <https://gns3.com/community/featured/advanced-port-name-formatting> few
> years ago.
> Looks like everyone agreed that it will be very coll feature to create
> more granular naming.
>
> I just want to propose adding some sort of calculations. IMHO this would
> be easier than changing GUI. Mo' better to create editor like in default
> switch menu of course.
>
> I.e. we all using vMX where first two ports are special purpose ports.
> What about adding calculation to make third port ge-0/0/0?
> ge-0/0/{port0-1} = ge-0/0/-1, ge-0/0/0
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GNS3/gns3-gui/discussions/3413>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APZX5I4U3ZRFUSG6PZ34UG3WNF4RBANCNFSM6AAAAAAS6ELFSA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Ean L. Towne

***@***.***

**Comment by ivanovuri** [comment]
Normally I add vMX by hand, so it will be better to have editing ability.
Nevertheless, any help how to do it?
After opening appliance file I see:
`
 "usage": "Connect VCP by port Eth1.\nData port ge/xe-x/0/0 to ge/xe-x/0/9 mapping to Eth3 to Eth12.\nInitial username is root, password is root.\n",
"symbol": "juniper-vmx.svg",
"first_port_name": "Eth0",
"port_name_format": "Eth{port1}",
`
P.S. Looks like appliance creator doesn't know how to make correct interfaces ;-)

**Comment by eantowne** [comment]
Later today I will dig out a sample and provide it. If you do not see any
response by tomorrow morning, please remind me.

On Wed, Dec 14, 2022 at 08:18 ivanovuri ***@***.***> wrote:

> Normally I add vMX by hand, so it will be better to have editing ability.
> Nevertheless, any help how to do it?
> After opening appliance file I see:
> "usage": "Connect VCP by port Eth1.\nData port ge/xe-x/0/0 to ge/xe-x/0/9
> mapping to Eth3 to Eth12.\nInitial username is root, password is root.\n",
> "symbol": "juniper-vmx.svg", "first_port_name": "Eth0", "port_name_format":
> "Eth{port1}",
> P.S. Looks like appliance creator doesn't know how to make correct
> interfaces ;-)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GNS3/gns3-gui/discussions/3413#discussioncomment-4400136>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APZX5IYSFVLJJPJN2DZKQT3WNHCJLANCNFSM6AAAAAAS6ELFSA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Ean L. Towne

***@***.***

**Comment by eantowne** [comment]
Take a look at the JSON specification for the .gns3a file found [HERE](https://github.com/GNS3/gns3-registry/blob/master/schemas/appliance_v6.json). Line 270 is probably what you are looking for. I had to dig around to find this as I seem to have misplaced some of my custom .gns3a files.

**Comment by eantowne** [comment]
I can understand not wanting to use the standard vMX appliance configuration. For vMX I created a custom .gns3a file for RE vs PFE so that I could create RE, RE0 and RE1, Also PFE0-4. This allows me to run multi-RE, multi-PFE vMX's. The standard appliance file does not support that.

I do agree that the ability to modify all of the attributes (like individual port names) would be great, however, I would say focus on the major functionality stuff first. Just my 2 cents worth probably way less than that ;)
